include/envoy/upstream/outlier_detection.h

```diff
@@ -82,7 +90,7 @@ class DetectorHostMonitor {
    *         -1 means that the host did not have enough request volume to calculate success rate
    *         or the cluster did not have enough hosts to run through success rate outlier ejection.
    */
-  virtual double successRate() const PURE;
+  virtual double successRate(envoy::data::cluster::v2alpha::OutlierEjectionType) const PURE;
```

Comment:
From an API perspective, the use of `OutlierEjectionType` here feels strange to me. Aren't you basically just splitting on external vs. internal and possibly returning a unified view if that is what is configured? Should we have a different enum that makes that more clear? Same elsewhere?

Reply:
Actually I used enum originally, but then refactored to use constants from API. Refactored back to enum.